lib/matplotlib/backends/backend_gtk3.py

```diff
@@ -9,7 +9,12 @@ def fn_name(): return sys._getframe(1).f_code.co_name
 try:
     import gi
 except ImportError:
-    raise ImportError("Gtk3 backend requires pygobject to be installed.")
+    try:
+        import pgi
+        pgi.install_as_gi()
```

Comment:
Well, this is better than `import pgi` secretly having side effects.  I am pretty sure that function is doing fun things to the path/import hooks.

On further consideration, this actually is a reasonable pattern.